Makefile

```diff
@@ -108,7 +108,7 @@ lint_python:
 	@$(PYTHON) -c 'print("Linting Python files...")'
 	@DOCKER_BUILDKIT=1 $(DOCKER) build --rm --target $@ -t $(IMAGE_NAME):latest-$@ .
 	@$(DOCKER) run --rm -v $$(pwd):/usr/src/app/ $(IMAGE_NAME):latest-$@
-	@$(FIND) . -type f -name '*.py' -exec $(DOCKER) run --rm -v $$(pwd):/data cytopia/black:latest --check {} +
+	@$(FIND) . -type f -name '*.py' -exec $(DOCKER) run --rm -v $$(pwd):/data cytopia/black@sha256:20af8eecc054b0bf321ff5cbaf2a3b4bab7611fb093620b42d61a866206c7b6e --check {} +
```

Comment:
This is the primary change, everything else is running black, updating pylint comments, and re-generating reports.

Reply:
We should probably make a task to revisit this later to see if the issue is fixed and if we can switch back to `latest`.